### PR TITLE
[feature/RPA-989] Create action Sharepoint Excel: Delete columns

### PR DESCRIPTION
**Pull request by marcin-depcik**
## Description

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-common/src/actions/actions.ts [hunk 1]
```diff
@@ -11,7 +11,8 @@ export enum CloudPath {
 export enum ActionRegex {
     VARIABLE_NAME = '^[a-zA-Z0-9_]*$',
     EXCEL_WORKSHEET_NAME = '^[^\\[\\]\\*\\?\\/\\\\\\:\\|]*$',
-    EXCEL_CELL_ADDRESS = '^[a-zA-Z]+\\$?\\d+'
+    EXCEL_CELL_ADDRESS = '^[a-zA-Z]+\\$?\\d+',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-common/src/actions/actions.ts]: imo this also should be '^[A-Z]+\\$?\\d+'

File: runbotics/runbotics-common/src/actions/actions.ts [hunk 2]
```diff
@@ -11,7 +11,8 @@ export enum CloudPath {
...
+    EXCEL_COLUMN_NAME = "^[A-Z]$",
```
Comment (star-in-the-jar) [inline runbotics/runbotics-common/src/actions/actions.ts]: it's not right. column can be also BAF but here can be only 1-letter

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/process/process.json [hunk 3]
```diff
@@ -326,6 +326,10 @@
     "Process.Details.Modeler.Actions.SharePointExcel.OpenFileFromRoot.FilePath.Info": "Refers to the location path, where the starting point is the folder, where the bot lands after login to the account, e.g. folderName/fileName.xlsx",
     "Process.Details.Modeler.Actions.SharePointExcel.OpenFileFromRoot.ListName.Info": "The default name is 'Documents'",
     "Process.Details.Modeler.Actions.CloudExcel.CloseSession.Label": "Close session",
+    "Process.Details.Modeler.Actions.CloudExcel.DeleteColumns.Label": "Delete columns",
+    "Process.Details.Modeler.Actions.CloudExcel.DeleteColumns.StartColumn": "Start column (value must be [A-Z])",
+    "Process.Details.Modeler.Actions.CloudExcel.DeleteColumns.EndColumn": "End column (value must be [A-Z])",
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/process/process.json]: In all excel desktop actions we have example value (and most of the restrictions) in the tooltip. 
+
[A-Z] contains regex expression which is not familiar to non-programmers. imo "uppercase letters A-Z" + example in the tooltip are enough

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/process/process.json [hunk 4]
```diff
@@ -326,6 +326,10 @@
...
+    "Process.Details.Modeler.Actions.CloudExcel.DeleteColumns.EndColumn.Info": "If not specified, only the starting column will be deleted",
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/process/process.json]: 👌

File: runbotics/runbotics-desktop/src/action/microsoft/excel/automation/cloud-excel.action-handler.ts [hunk 5]
```diff
@@ -59,6 +59,19 @@ export class CloudExcelActionHandler extends StatefulActionHandler {
         return this.excelService.setRange(this.session, input.range, input.values);
     }
 
+    deleteColumns(input: SharepointTypes.CloudExcelDeleteColumnsActionInput) {
+        const startColumn = input.startColumn.match(ActionRegex.EXCEL_COLUMN_NAME);
+        const endColumn = input.endColumn ? input.endColumn.match(ActionRegex.EXCEL_COLUMN_NAME) : startColumn;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/automation/cloud-excel.action-handler.ts:68]: right now this won't work for e.g. range AB-AG

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts [hunk 6]
```diff
@@ -179,6 +179,20 @@ export class ExcelService {
...
+    /**
+     * @see https://learn.microsoft.com/en-us/graph/api/range-delete?view=graph-rest-1.0&tabs=http
+     */
+    deleteColumns(session: ExcelSession, columnRange: string) {
+        return this.microsoftGraphService.
+            post(
+                this.createWorkbookUrl(session, `/worksheets/${session.worksheetName}/range(address='${columnRange}')/delete`),
+                {
+                    ...this.getSessionHeader(session),
+                    shift: 'left'
```
Comment (star-in-the-jar) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts:202]: what is it for?

**Review by star-in-the-jar — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-common/src/actions/actions.ts (hunk 1 above)
Comment (preludium) [reply runbotics/runbotics-common/src/actions/actions.ts]: why dollar sign? and imo there is no difference between lower and upper case

**Review by preludium — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-common/src/actions/actions.ts (hunk 1 above)
Comment (marcin-depcik) [reply runbotics/runbotics-common/src/actions/actions.ts]: I think this pattern is completely wrong. It verifies the coordinates of a cell, e.g. C1, but also CaB123, so imo it should match a string with two substrings. The first part matches letters and the second part matches numbers. My suggestion is a pattern that looks as follows: ^[a-zA-Z]+(\d+)$

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts (hunk 6 above)
Comment (marcin-depcik) [reply runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts:202]: This function sends request to delete column/columns in worksheet. You can read more here: https://learn.microsoft.com/en-us/graph/api/range-delete?view=graph-rest-1.0&tabs=http

File: runbotics/runbotics-common/src/actions/actions.ts (hunk 1 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-common/src/actions/actions.ts]: I'd only added '+' after [A-Z].  What @marcin-depcik suggests is best solution but imo without the brackets: ^[a-zA-Z]+\d+$. Unless you want to immediately split letters and numbers which sounds like a good idea, but then imo both should be in brackets ^([a-zA-Z]+)(\d+)$

Comment (star-in-the-jar) [reply runbotics/runbotics-common/src/actions/actions.ts]: > why dollar sign? and imo there is no difference between lower and upper case

the dollar sign means end of line

Comment (marcin-depcik) [reply runbotics/runbotics-common/src/actions/actions.ts]: I mean the address 'CaB123' should be valid input, correct me if I am wrong.

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/process/process.json [hunk 7]
```diff
@@ -330,6 +330,11 @@
     "Process.Details.Modeler.Actions.SharePointExcel.OpenFileFromRoot.FilePath.Info": "Odnosi się do ścieżki, gdzie punktem początkowym jest folder, w którym bot ląduje po zalogowaniu się do konta, np. nazwaFolderu/nazwaPliku.xlsx",
     "Process.Details.Modeler.Actions.SharePointExcel.OpenFileFromRoot.ListName.Info": "Domyślna nazwa to 'Documents'",
     "Process.Details.Modeler.Actions.CloudExcel.CloseSession.Label": "Zakończ sesję",
+    "Process.Details.Modeler.Actions.CloudExcel.DeleteColumns.Label": "Usuń kolumny",
+    "Process.Details.Modeler.Actions.CloudExcel.DeleteColumns.StartColumn": "Kolumna początkowa",
+    "Process.Details.Modeler.Actions.CloudExcel.DeleteColumns.EndColumn": "Kolumna końcowa",
+    "Process.Details.Modeler.Actions.CloudExcel.DeleteColumns.StartColumn.Info": "Nazwa kolumny początkowej (np. A)",
+    "Process.Details.Modeler.Actions.CloudExcel.DeleteColumns.EndColumn.Info": "Nazwa kolumny końcowej (np. H), jeśli nie określono wartości, usunięta zostanie tylko kolumna początkowa",
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/process/process.json]: desktop excel examples are in form ..., np. sth - no brackets, yes comma
+
grammar suggestion:
Nazwa kolumny końcowej, np. H. Jeżeli nie określono wartości, usunięta zostanie tylko kolumna początkowa

**Review by star-in-the-jar — APPROVED** [review]

**Review by preludium — APPROVED** [review]

**Pull request merged**